From: Haifeng Xu <haifeng.xu@shopee.com>
Subject: mm/memory_hotplug: remove reset_node_managed_pages() in hotadd_init_pgdat()
Date: Wed, 7 Jun 2023 02:45:48 +0000

managed pages has already been set to 0 in free_area_init_core_hotplug(),
via zone_init_internals() on each zone.  It's pointless to reset again.

Furthermore, reset_node_managed_pages() no longer needs to be exposed
outside of mm/memblock.c.  Remove declaration in include/linux/memblock.h
and define it as static.

In addtion to this, the only caller of reset_node_managed_pages() is
reset_all_zones_managed_pages(), which is annotated with __init, so it
should be safe to also mark reset_node_managed_pages() as __init.

Link: https://lkml.kernel.org/r/20230607024548.1240-1-haifeng.xu@shopee.com
Signed-off-by: Haifeng Xu <haifeng.xu@shopee.com>
Suggested-by: David Hildenbrand <david@redhat.com>
Cc: Michal Hocko <mhocko@kernel.org>
Cc: Mike Rapoport (IBM) <rppt@kernel.org>
Cc: Oscar Salvador <osalvador@suse.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/memblock.h |    1 -
 mm/memblock.c            |    2 +-
 mm/memory_hotplug.c      |    1 -
 3 files changed, 1 insertion(+), 3 deletions(-)

--- a/include/linux/memblock.h~mm-memory_hotplug-remove-reset_node_managed_pages-in-hotadd_init_pgdat
+++ a/include/linux/memblock.h
@@ -128,7 +128,6 @@ int memblock_clear_nomap(phys_addr_t bas
 
 void memblock_free_all(void);
 void memblock_free(void *ptr, size_t size);
-void reset_node_managed_pages(pg_data_t *pgdat);
 void reset_all_zones_managed_pages(void);
 
 /* Low level functions */
--- a/mm/memblock.c~mm-memory_hotplug-remove-reset_node_managed_pages-in-hotadd_init_pgdat
+++ a/mm/memblock.c
@@ -2122,7 +2122,7 @@ static unsigned long __init free_low_mem
 
 static int reset_managed_pages_done __initdata;
 
-void reset_node_managed_pages(pg_data_t *pgdat)
+static void __init reset_node_managed_pages(pg_data_t *pgdat)
 {
 	struct zone *z;
 
--- a/mm/memory_hotplug.c~mm-memory_hotplug-remove-reset_node_managed_pages-in-hotadd_init_pgdat
+++ a/mm/memory_hotplug.c
@@ -1194,7 +1194,6 @@ static pg_data_t __ref *hotadd_init_pgda
 	 * online_pages() and offline_pages().
 	 * TODO: should be in free_area_init_core_hotplug?
 	 */
-	reset_node_managed_pages(pgdat);
 	reset_node_present_pages(pgdat);
 
 	return pgdat;
_
